pandas/core/series.py

```diff
@@ -1522,40 +1555,157 @@ def unique(self):
+        ...            for _ in range(3)]).unique(raw=False)
+        0   2016-01-01
+        dtype: datetime64[ns]
+
+        as well as timezones:
 
         >>> pd.Series([pd.Timestamp('2016-01-01', tz='US/Eastern')
-        ...            for _ in range(3)]).unique()
-        array([Timestamp('2016-01-01 00:00:00-0500', tz='US/Eastern')],
-              dtype=object)
+        ...            for _ in range(3)]).unique(raw=False)
+        0   2016-01-01 00:00:00-05:00
+        dtype: datetime64[ns, US/Eastern]
 
         An unordered Categorical will return categories in the order of
         appearance.
 
-        >>> pd.Series(pd.Categorical(list('baabc'))).unique()
+        >>> pd.Series(pd.Categorical(list('baabc'))).unique(raw=False)
+        0    b
+        1    a
+        4    c
+        dtype: category
+        Categories (3, object): [b, a, c]
+
+        >>> pd.Series(pd.Categorical(list('baabc'))).unique(raw=True)
         [b, a, c]
         Categories (3, object): [b, a, c]
 
         An ordered Categorical preserves the category ordering.
 
         >>> pd.Series(pd.Categorical(list('baabc'), categories=list('abc'),
-        ...                          ordered=True)).unique()
-        [b, a, c]
+        ...                          ordered=True)).unique(raw=False)
+        0    b
+        1    a
+        4    c
+        dtype: category
         Categories (3, object): [a < b < c]
-        """
-        result = super(Series, self).unique()
 
-        if is_datetime64tz_dtype(self.dtype):
-            # we are special casing datetime64tz_dtype
-            # to return an object array of tz-aware Timestamps
+        As an example for dealing with `return_inverse`, we consider the
+        following example (the reason we use a non-default index is only for
+        demonstration purposes, because this is also something the inverse
+        needs to reconstruct):
+
+        >>> animals = pd.Series(['lama', 'cow', 'lama', 'beetle', 'lama'],
+        ...                     index=[1, 4, 9, 16, 25])
+
+        >>> animals_unique, inverse = animals.unique(raw=False,
+        ...                                          return_inverse=True)
+        >>> animals_unique
+        1       lama
+        4        cow
+        16    beetle
+        dtype: object
+
+        >>> inverse
+        1      1
+        4      4
+        9      1
+        16    16
+        25     1
+        dtype: int64
+
+        This can be used to reconstruct the original object from its unique
+        values as follows
+
+        >>> reconstruct = animals_unique.reindex(inverse)
+        >>> reconstruct
+        1       lama
+        4        cow
+        1       lama
+        16    beetle
+        1       lama
+        dtype: object
+
+        We see that the values of `animals` get reconstructed correctly, but
+        the index does not match yet  -- consequently, the last step is to
+        correctly set the index.
```

Comment:
At this point, it's be really neat to have #22225 available to use `.set_index`...